### PR TITLE
repl: add quick command

### PR DESCRIPTION
**Pull request by regomne**
The feature "quick command" in frida REPL, like:

```
[Local::a.exe ]-> .d 401000 10
           0  1  2  3  4  5  6  7  8  9  A  B  C  D  E  F  0123456789ABCDEF
00401000  55 8b ec 81 c4 fc fe ff ff 8b 45 0c 83 f8 01 75  U.........E....u
```

The quick cmd starts with '.' (same as nodejs REPL), and can be customized dynamically by the javascript loaded.

For example, set a new quick cmd 'u' in `agent.js` :

```js
REPL.setQuickCmd(
    'u', // the quick cmd
    1, // at least 1 argument
    function disasm(addr, inst_cnt = 10) { ... } // target function
)
```

Which can be used in REPL:

```
[Local::a.exe ]-> .u 401000
00401000    push ebp
00401001    mov ebp, esp
00401003    add esp, 0xfffffefc
00401009    mov eax, dword ptr [ebp + 0xc]
0040100C    cmp eax, 1
0040100F    jne 0x401028
00401011    push dword ptr [0x403864]
00401017    push 1
00401019    push 0x466
0040101E    push dword ptr [ebp + 8]
```

### CODE REVIEW

**Comment by regomne** [comment]
@oleavr added REPL class support to repl.py

**Comment by regomne** [comment]
@oleavr Hey guy, what about this PR?

File: frida_tools/repl.py [hunk 1]
```diff
@@ -509,6 +513,17 @@ def _do_magic(self, statement):
...
+        
+        def _do_quick_cmd(self, statement):
+            tokens = shlex.split(statement)
+            if len(tokens) == 0:
+                self._print("Invalid quick cmd")
+                return
+
+            expression = "REPL._doQuickCmd('{}')".format(encode_uri_component(json.dumps(tokens)))
+
+            if not self._eval_and_print(expression):
```
Comment (oleavr) [inline frida_tools/repl.py:525]: Should define an RPC method to call instead.

File: frida_tools/repl.py [hunk 2]
```diff
@@ -629,6 +644,33 @@ def _make_repl_runtime(self, indent):
...
 
+class REPL {
+    #cmdDispatcher;
+    constructor() {
+        this.#cmdDispatcher = new Map();
+    }
+    _doQuickCmd(s) {
+        const args = JSON.parse(decodeURIComponent(s));
+        const cmd = args[0];
+        const func_info = this.#cmdDispatcher.get(cmd)
```
Comment (oleavr) [inline frida_tools/repl.py:655]:
```suggestion
        const funcInfo = this.#cmdDispatcher.get(cmd)
```

(Here and elsewhere:) Variables should always be `camelCase`.

**Review by oleavr — CHANGES_REQUESTED** [review]
Sorry for the delay here, haven't been able to keep on top of my notifications -- will keep an extra eye on this one going forward 😊

File: frida_tools/repl.py (hunk 2 above)
Comment (regomne) [reply frida_tools/repl.py]: Modified with RPC using and camelCase used.

**Comment by oleavr** [comment]
Thanks! :fire: Merged with some minor tweaks. (Hopefully the changes make sense; if not feel free to open another PR!)

**Pull request merged**